tests_and_analysis/test/euphonic_test/test_qpoint_phonon_modes.py

```diff
@@ -544,19 +550,49 @@ def test_calculate_dos_from_phonopy(
              'quartz_554_full_mode_widths.json',
              'quartz_554_full_adaptive_dos.json',
              np.arange(0, 155, 0.1)*ureg('meV'))])
-    def test_calculate_dos_with_mode_widths(
+    def test_calculate_dos_with_mode_widths_ref_method(
             self, material, qpt_ph_modes_json, mode_widths_json,
             expected_dos_json, ebins):
         qpt_ph_modes = get_qpt_ph_modes_from_json(material, qpt_ph_modes_json)
-        with open(os.path.join(get_fc_dir(), mode_widths_json), 'r') as fp:
-            modw_dict = json.load(fp)
-        mode_widths = modw_dict['mode_widths']*ureg(
-            modw_dict['mode_widths_unit'])
+        mode_widths = get_mode_widths(mode_widths_json)
         dos = qpt_ph_modes.calculate_dos(
-            ebins, mode_widths=mode_widths)
+            ebins, mode_widths=mode_widths, adaptive_method='reference')
         expected_dos = get_expected_spectrum1d(expected_dos_json)
         check_spectrum1d(dos, expected_dos)
 
+    @pytest.mark.parametrize(
+        ('material, qpt_ph_modes_json, mode_widths_json, '
+         'expected_dos_json, ebins'), [
+            ('quartz', 'quartz_554_full_qpoint_phonon_modes.json',
+             'quartz_554_full_mode_widths.json',
+             'quartz_554_full_adaptive_dos_fast.json',
+             np.arange(0, 155, 0.1)*ureg('meV'))])
+    def test_calculate_dos_with_mode_widths_fast_method(
+            self, material, qpt_ph_modes_json, mode_widths_json,
+            expected_dos_json, ebins):
+        qpt_ph_modes = get_qpt_ph_modes_from_json(material, qpt_ph_modes_json)
+        mode_widths = get_mode_widths(mode_widths_json)
+        dos = qpt_ph_modes.calculate_dos(
+            ebins, mode_widths=mode_widths, adaptive_method='fast')
+        expected_dos = get_expected_spectrum1d(expected_dos_json)
+        check_spectrum1d(dos, expected_dos, 1e-13)
```

Comment:
Unless I'm missing something this can also be tested in the same test as `test_calculate_dos_with_mode_widths_ref_method` (formerly `test_calculate_dos_with_mode_widths`) above if another parameter is added.